pkg/kubectl/resource/visitor.go

```diff
@@ -224,17 +225,60 @@ type URLVisitor struct {
 }
 
 func (v *URLVisitor) Visit(fn VisitorFunc) error {
-	res, err := http.Get(v.URL.String())
+	body, err := v.readHttpWithRetries(httpgetImpl, time.Second, v.URL.String())
 	if err != nil {
 		return err
 	}
-	defer res.Body.Close()
-	if res.StatusCode != 200 {
-		return fmt.Errorf("unable to read URL %q, server reported %d %s", v.URL, res.StatusCode, res.Status)
+	defer body.Close()
+	v.StreamVisitor.Reader = body
+	return v.StreamVisitor.Visit(fn)
+}
+
+// readHttpWithRetries tries to http.Get the v.URL 3 times before giving up.
+func (v *URLVisitor) readHttpWithRetries(get httpget, duration time.Duration, u string) (io.ReadCloser, error) {
+	var err error
+	var body io.ReadCloser
+	for i := 0; i < 3; i++ {
```

Comment:
I prototyped my understanding of this suggestion.  It makes the code look less hacky, but I think this is mainly because it obfuscates that it is actually hardcoded to a const.  If we plan to expose this in the future it makes sense.  It could also be indication to the caller that the http requests will be retried.  I don't have a strong opinion about this so I will push the changes to see what you think about it.